src/services/codefixes/importFixes.ts

```diff
@@ -154,8 +154,13 @@ namespace ts.codefix {
         // We sort the best codefixes first, so taking `first` is best for completions.
         const moduleSpecifier = first(getNewImportInfos(program, sourceFile, exportInfos, host, preferences)).moduleSpecifier;
         const fix = first(getFixForImport(exportInfos, symbolName, symbolToken, program, sourceFile, host, preferences));
-        return { moduleSpecifier, codeAction: codeActionForFix({ host, formatContext }, sourceFile, symbolName, fix, getQuotePreference(sourceFile, preferences)) };
+        return { moduleSpecifier, codeAction: codeFixActionToCodeAction(codeActionForFix({ host, formatContext }, sourceFile, symbolName, fix, getQuotePreference(sourceFile, preferences))) };
     }
+
+    function codeFixActionToCodeAction({ description, changes, commands }: CodeFixAction): CodeAction {
+        return { description, changes, commands };

```

Comment:
Drop `commands` too?

Reply:
Currently `protocol.ts` `CompletionEntryDetails` contains a `CodeAction[]` and `CodeAction` has `commands` on it. I could make a `CodeActionWithoutCommands` interface if you think it's important that a completion never has commands (which it probably shouldn't since they're easy to trigger on accident).

Reply:
@amcasey Please open a new issue if you would like `CodeActionWithoutCommands`.

Reply:
Not very important.  Thanks for following up though!